### PR TITLE
Implementing a possibility to download system/environment information

### PR DESCRIPTION
**Pull request by Radek-Suski**
Implementing a possibility to download system/environment information for support purposes.
The output is a text file which is probably not the best solution. Hope for some input. 
It adds a button in "administrator/index.php?option=com_admin&view=sysinfo" which allows to create a text file with all relevant settings.
All sensitive informations should have been removed.

### CODE REVIEW

**Comment by brianteeman** [comment]
Awesome stuff. Can you alpha order the new language string please

**Comment by Radek-Suski** [comment]
Sure. I will have to add some fixes to make Travis happy. Will try to fix it tomorrow. 
But I think it can be already tested as it is as the functionality is provided.

**Comment by brianteeman** [comment]
In the file section the output is
administrator/components: administrator/components is writable
Shouldnt it be
administrator/components:  is writable

Easier to read

**Comment by Radek-Suski** [comment]
Yep. Thank for the hint. Changing now

**Comment by Radek-Suski** [comment]
Ok. I think I am done and Travis is happy. Thanks in advance for help.

**Comment by zero-24** [comment]
@Radek-Suski I have just open a PR that should fix CS errors :smile: https://github.com/Radek-Suski/joomla-cms/pull/1

Maybe you can have a look into the doc block lines with `...` and fill the information there?

I will have a look into the next week!

Thanks a lot!

**Comment by Radek-Suski** [comment]
I just merged your request. Will take a look at the docblocks tomorrow. 
It would be nice if someone could test it

**Comment by brianteeman** [comment]
It does what it says but do you think it is really needed or useful to output all of the phpinfo 

Would it also be useful to add a list of ALL installed (non-core) extensions and their version numbers
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by Radek-Suski** [comment]
@brianteeman I am not entirely sure if the entire PHP info is needed. It's the reason I asked for feedback.I was wondering about installed extensions as well. Not sure if it wouldn't be a huge informations disclosure.

**Comment by brianteeman** [comment]
If the generated file is only ever downloaded and not stored on the server
(which it is) then I dont see where the information disclosure would be. A
super user has full access to that information anyway

On 8 February 2015 at 21:52, Radek Suski <notifications@github.com> wrote:

> @brianteeman <https://github.com/brianteeman> I am not entirely sure if
> the entire PHP info is needed. It's the reason I asked for feedback.I was
> wondering about installed extensions as well. Not sure if it wouldn't be a
> huge informations disclosure.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6021#issuecomment-73434262>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by Radek-Suski** [comment]
I was thinking about people posting the file in some forums (it is the main purpose) and disclosing some (maybe) relevant information.
I am not sure however if this may be somehow critical.

**Comment by brianteeman** [comment]
Take a look at the information created by the FPA
https://github.com/ForumPostAssistant/FPA/

On 8 February 2015 at 22:06, Radek Suski <notifications@github.com> wrote:

> I was thinking about people posting the file in some forums (it is the
> main purpose) and disclosing some (maybe) relevant information.
> I am not sure however if this may be somehow critical.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6021#issuecomment-73434960>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by zero-24** [comment]
@Radek-Suski On testing all looks ok here. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by zero-24** [comment]
@Radek-Suski do you think we should add a text bevor the block like 
```
=============
PhpSettings
=============
```

**Comment by Radek-Suski** [comment]
It depend if we want to have those settings included or not. 
I agree with @brianteeman that a list of installed extensions may be more important. 
Not sure though

**Comment by phproberto** [comment]
FFS I just noticed you did a wrong PR with the code of this in the other issue :question: 

See:
https://github.com/joomla/joomla-cms/pull/5647

I thought this was what you were trying to do there :dash:

**Comment by phproberto** [comment]
I have submitted a PR against your repo @Radek-Suski :
https://github.com/Radek-Suski/joomla-cms/pull/2

**Comment by phproberto** [comment]
It would be interesting to rebase your branch against latest staging. The staging `sysinfo` model has been changed so it can save us issues (I had to redo this patch 3 times to avoid causing conflicts with that).

**Comment by MAT978** [comment]
@test 

works like a charm and fabulous improvement!

thanks <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by MAT978** [comment]
@phproberto 

At this point, I'm not able to see the "download as text" option?
Maybe we have to wait @Radek-Suski to merge your PR?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by msdevcon** [comment]
@Test
works fine! thanks!

**Comment by Chalkin** [comment]
@test
Download of the file works fine! Information is available. Looks good!
Tested with Joomla 3.4.1-dev
Chrome 41.0.2272.89 MacOS<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by ceus1984** [comment]
Succsessfully tested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by brianteeman** [comment]
Multiple good tests - setting RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by Radek-Suski** [comment]
I want to add list of installed extensions into this. But I don't want to parse al XML files. Any idea for a better solution. I can't locate the right method in the com_installer. I can debug it through but if someone know the right place I would be thankful for an advice.

**Comment by brianteeman** [comment]
Why not query the _extensions table

**Comment by Radek-Suski** [comment]
It doesn't contain the version of the installed extension

**Comment by mbabker** [comment]
The manifest column cache has it, you'd have to json_decode it.

**Comment by Radek-Suski** [comment]
Ah. Cool. Thank @mbabker

**Comment by mbabker** [comment]
Reset to pending as another "feature" was added.  Please re-test.

@Radek-Suski GitHub shows the branch can't be automatically merged, could you please merge the current staging branch in to your branch?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by MAT978** [comment]
What I was asking above ;)

@test 

Both 'Download as JSON' and 'Download as text' work fine here.

Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by sabindice** [comment]
@test both download works<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

**Comment by zero-24** [comment]
Thanks for testing. Moving to RTC again :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6021">issues.joomla.org/joomla-cms/6021</a>.</sub>

File: administrator/components/com_admin/models/sysinfo.php [hunk 1]
```diff
@@ -203,6 +362,76 @@ public function &getPHPInfo()
...
+	 *
+	 * @since  3.5
+	 */
+	public function getExtensions()
+	{
+		$installed = array();
+		$db = JFactory::getDbo();
+		$query = $db->getQuery(true)
+			->select('*')
+			->from('#__extensions');
```
Comment (roland-d) [inline administrator/components/com_admin/models/sysinfo.php]: Please use the $db->quoteName() around the table name.

**Comment by wilsonge** [comment]
Hey @Radek-Suski . Can you do things:
a) Fix the issue @roland-d mentioned about quoteName
b) Merge in staging so that I can get this awesome feature merged in!
Thanks!

**Comment by wilsonge** [comment]
Hey @Radek-Suski - Just a gentle reminder on this one now JAB is over :)

**Comment by phproberto** [comment]
I have created a new PR with this code rebased to be sure we don't have to deal with the sysinfo model conflicts:

https://github.com/joomla/joomla-cms/pull/7129

**Comment by Radek-Suski** [comment]
@phproberto So you are taking over?

**Comment by phproberto** [comment]
Sorry I thought it was ready to merge after adding the extensions list.

My PR just reorders the commits so the other core sysinfo patch is applied before yours and not conflicting anymore. Commits are the same.

I thought I was helping but I'll close my PR. If you need help ping me ;)

**Comment by Radek-Suski** [comment]
I am perfectly fine with doing it Roberto. It wasn't an accusation, just a question :)

**Comment by phproberto** [comment]
Ok :)

So let's close this and I'll reopen #7129

**Pull request closed without merging**